### PR TITLE
`cats.Defer` instance for `Stream`.

### PR DESCRIPTION
**Pull request by oskin1**

### CODE REVIEW

File: core/shared/src/main/scala/fs2/Stream.scala [hunk 1]
```diff
@@ -4973,6 +4973,12 @@ object Stream extends StreamLowPriority {
       def empty[A]: Stream[F, A] = Stream.empty
       def combineK[A](x: Stream[F, A], y: Stream[F, A]): Stream[F, A] = x ++ y
     }
+
+  /** `Defer` instance for `Stream` */
+  implicit def deferInstance[F[_]]: Defer[Stream[F, *]] =
+    new Defer[Stream[F, *]] {
+      override def defer[A](fa: => Stream[F, A]): Stream[F, A] = Stream(()) >> fa
```
Comment (djspiewak) [inline core/shared/src/main/scala/fs2/Stream.scala:4980]: Wouldn't it be a bit faster to do `Stream.empty ++ fa`?

Comment (oskin1) [reply core/shared/src/main/scala/fs2/Stream.scala]: Makes sense, thanks!

**Comment by mpilquist** [comment]
Thanks!

**Comment by mpilquist** [comment]
BTW, we could do same for `Pull`

**Pull request merged**